task12-whack-a-mole/src/game.js

```diff
@@ -41,16 +55,60 @@ function Game() {
         scoreBoard.textContent = 0;
         timeUp = false;
         score = 0;
+        if(document.getElementById("name").value === "") name = "Anonymous"
+        else name = document.getElementById("name").value;
         this.peep();
-        setTimeout(() => timeUp = true, 10000)
+        setTimeout(() => {
+          timeUp = true;
+          this.changeScore();
+          this.levelUp();
+        }, 10000)
+    }
+
+    this.levelUp = () => {
+      if(score > 5 && minPeepTime >= 500) {
+        minPeepTime -= 500;
+        level++;
+        levelText.textContent = level;   
+      }
     }
 
     this.bonk = e => {
         if (!e.isTrusted) return;
-        score++;
+        if(!timeUp) {
+          audioSuccess.play();
+          score++;
+          scoreBoard.textContent = score;
+        };
         e.target.classList.remove('up');
-        scoreBoard.textContent = score;
+    }
+
+    this.sound = e => {
+      if(!e.target.classList.contains('mole') && !timeUp) audioFail.play();
```

Comment:
Wrap the body of the `if` clause in `{}` to make it clearer to read or take advantage of short circuit and implement `!e.target.classList.contains('mole') && !timeUp && audioFail.play()`